src/lightning/fabric/wrappers.py

```diff
@@ -238,6 +258,167 @@ def __setattr__(self, name: str, value: Any) -> None:
             super().__setattr__(name, value)
 
 
+class _BackwardTensor(torch.Tensor):
+    """Tensor wrapper subclass that allows us to hook the strategy machinery into ``loss.backward()``.
+
+    The implementation is inspired by:
+        - https://pytorch.org/docs/main/notes/extending.html#operations-on-multiple-types-that-define-torch-function
+        - :class:`~torch.distributed._tensor.api.DTensor`
+        - https://github.com/albanD/subclass_zoo/blob/08dfcf9/custom_parameter.py
+        - https://github.com/albanD/subclass_zoo/issues/29
+        - https://github.com/pytorch/pytorch/blob/ba86dfcd83/torch/testing/_internal/two_tensor.py
+
+    """
+
+    def __new__(cls, tensor: torch.Tensor, strategy: Strategy, module: torch.nn.Module) -> "_BackwardTensor":
+        wrapper = torch.Tensor._make_wrapper_subclass(
```

Comment:
I am not sure about using protected func which could be subject to change between one version to another

Reply:
We do not offer future compatibility with PyTorch. It's a best effort thing